---
title: "Supplement: Explicit Scale Simulation for analysis of RNA-sequencing with ALDEx2
"
shorttitle: "Scale ALDEx2 Supp"
citeproc: TRUE
cite-method: citeproc
bibliography: /Users/ggloor/Library/texmf/bibtex/bib/bibdesk_refs.bib
csl: /Users/ggloor/Documents/0_git/csl_styles/nucleic-acids-research.csl
pdf-engine: latex
output:
  pdf_document
header-includes:
  \renewcommand{\figurename}{Supplementary Figure}
---

```{r echo=F, warning=F, message=F}
library(seqgendiff)
library(ALDEx2)
data(selex)
library(DESeq2)
library(edgeR)
```
```{r echo=FALSE, include=TRUE, out.width="0.8\\linewidth",  fig.align="center", fig.cap=c("The aldex.senAnalysis() function was used with the yeast transcriptome dataset to generate outputs with \\(\\gamma\\) values of 1e-3, 0.1, 0.2, 0.3, 0.4,0.5, 0.7 and 1. The plotGamma() function was used to plot the result. Transcripts that are statistically significant are shown in black, and if not significant are in grey. The \\(\\gamma\\) values, standardized effect size and number of significant transcripts at each value are given on the axes.  ")}

####
# Data generated by yst-sens.R
####

knitr::include_graphics("analysis/scaleSim-yeast.pdf")
## plotGamma needs to be cleaned up for production
## so this figure was done by hand with the example
## snippet below - ugly sorry
#plotGamma(x.sens, blackWhite=T)
#text(6.2,345, label=length(which(x.sens[[8]]$we.eBH < 0.05)))
#text(-2,345, label=length(which(x.sens[[7]]$we.eBH < 0.05)))
#text(-6.9,345, label=length(which(x.sens[[6]]$we.eBH < 0.05)))
#text(-9.6,345, label=length(which(x.sens[[5]]$we.eBH < 0.05)))
#text(-12.2,345, label=length(which(x.sens[[4]]$we.eBH < 0.05)))
#text(-14.8,345, label=length(which(x.sens[[3]]$we.eBH < 0.05)))
#text(-17.4,345, label=length(which(x.sens[[2]]$we.eBH < 0.05)))
#text(-20,345, label=length(which(x.sens[[1]]$we.eBH < 0.05)))
#text(-6.9,355, label="N significant")
```

Supplementary Figure 1 shows that  root cause of the large number of significant parts in this dataset is the very low dispersion of transcripts. Here we see  a graphical output from the *aldex.scaleSim()* function with the yeast transcriptome dataset described in the text [@Gierlinski:2015aa;@Schurch:2016aa]. This allows us to examine which transcripts are sensitive to even minimal amounts of scale uncertainty using \(\gamma = (1e-3, 0.1, 0.2, 0.3, 0.4, 0.5, 0.7, 1) \). Adding scale uncertainty increases the minimum dispersion in the analysis. Here it is obvious that even a negligible amount of scale uncertainty \(\gamma=0.1\) removes over half of the transcripts that were formerly significantly different, and all of these had very small effect sizes. 

We recommend a minimum scale uncertainty of between 0.2 and 0.5, and suggest that the *aldex.senAnalysis()* function be run on all datasets to identify transcripts that are sensitive to small \(\gamma\) values. The individual *aldex()* outputs can be accessed as sequential entries in the list output, or the analysis as a whole can be plotted with the *plotGamma* function.

```{r, brca-sens, echo=FALSE, include=TRUE, fig.dim=c(8,8),  fig.align="center", out.width="0.8\\linewidth",  fig.align="center", fig.cap=c('Adding scale to the BRCA dataset reduces false positives in the same way as the PD-1 dataset.')}

source('code/brca.R')

par(mfrow=c(2,2))
plot(means$coef[ald.row], means$FDR[ald.row], ylim=c(0,0.6), type='b',lwd=3,
  xlab="modeled minimum difference", ylab="mean FDR", col='blue')
title("A: FDR", adj=0, line= 0.8) 
points(means$coef[ald2.row], means$FDR[ald2.row],  type='b', lwd=3, 
  col='grey')
points(means$coef[ald5.row], means$FDR[ald5.row],  type='b', lwd=3, 
  col='grey50')
points(means$coef[des.row], means$FDR[des.row],  type='b', lwd=3, col='red')
points(means$coef[des5.row], means$FDR[des5.row],  type='b', lwd=3,
  col='darkred')
legend(0,0.6, legend=c('A 0', 'A 0.2', 'A 0.5'), pch=19,
  col=c('blue', 'grey', 'grey50'), bty="n", cex=1.2) 
legend(0.25,0.6, legend=c('D 0', 'D 0.5'), pch=19,
  col=c('red', 'darkred'), bty="n", cex=1.2) 
rect(xleft=0, xright=0.55,  ybottom=0.4, ytop=0.6)

plot(means$coef[ald.row], means$SEN[ald.row], ylim=c(0,1), lwd=3, type='b',
  xlab="modeled minimum difference", ylab="mean Sensitivity", col='blue')
title("B: sensitivity", adj=0, line= 0.8) 
points(means$coef[ald2.row], means$SEN[ald2.row],  lwd=3, type='b')
points(means$coef[ald2.row], means$SEN[ald2.row],  lwd=3, type='b', col='grey')
points(means$coef[ald5.row], means$SEN[ald5.row],  lwd=3, type='b', col='grey50')
points(means$coef[des.row], means$SEN[des.row],  lwd=3, type='b', col='red')
points(means$coef[des5.row], means$SEN[des5.row], lwd=3,  type='b', col='darkred')

addmin <- min(brca.data.out[[10]]$ald0$we.eBH[brca.data.out[[10]]$ald0$we.eBH > 0])/10

aldex.plot(brca.data.out[[10]]$ald0, type='volcano')
title("C: Volcano", adj=0, line= 0.8) 
points(brca.data.out[[10]]$ald0$diff.btw[which(brca.data.out[[10]]$ald2$we.eBH < 0.05)], 
  -log10(brca.data.out[[10]]$ald0$we.eBH[which(brca.data.out[[10]]$ald2$we.eBH < 0.05)]+ addmin), cex=0.6, pch=19, 
  col='orange')
points(brca.data.out[[10]]$ald0$diff.btw[which(brca.data.out[[10]]$ald5$we.eBH < 0.05)], 
  -log10(brca.data.out[[10]]$ald0$we.eBH[which(brca.data.out[[10]]$ald5$we.eBH < 0.05)]+ addmin), col='blue', 
  cex=0.6, pch=19)

abline(v=c(-0.5,0.5), lty=2, col='darkgrey')
abline(v=c(-1.5,1.5), lty=2, col='darkgrey')

gam0 <- expression(paste(gamma,"=0"))
gam2 <- expression(paste(gamma,"=0.2"))
gam5 <- expression(paste(gamma,"=0.5"))

legend(-6, 120, legend=c("NS", gam0, gam2, gam5), pch=19, col=c("black", "red", "orange", "blue"), cex=1.2)

aldex.plot(brca.data.out[[10]]$ald0, type='MW')
title("D: effect", adj=0, line= 0.8) 
points(brca.data.out[[10]]$ald0$diff.win[which(brca.data.out[[10]]$ald2$we.eBH < 0.05)], 
  (brca.data.out[[10]]$ald0$diff.btw[which(brca.data.out[[10]]$ald2$we.eBH < 0.05)]), cex=0.6, pch=19, 
  col='orange')
points(brca.data.out[[10]]$ald0$diff.win[which(brca.data.out[[10]]$ald5$we.eBH < 0.05)], 
  (brca.data.out[[10]]$ald0$diff.btw[which(brca.data.out[[10]]$ald5$we.eBH < 0.05)]), col='blue', 
  cex=0.6, pch=19)

abline(v=c(-0.5,0.5), lty=2, col='darkgrey')
abline(v=c(-1.5,1.5), lty=2, col='darkgrey')

```

<!-- # this is no longer useful

```{r, echo=F, warning=F, message=F,comment=F, result=F} 
#ALDEx2
load(file="analysis/yst.all.Rda")
load(file="analysis/yst.s.all.Rda")
load(file="analysis/yst.1.all.Rda")
# with gamma = 0.5
load('analysis/x.s.all.Rda')
load('analysis/yst.sens.Rda')
nsig <- x.sens[[2]]$we.eBH > 0.05 & x.sens[[1]]$we.eBH < 0.05

# data to compare different scale means
load(file='analysis/xt.lv.Rda')
load(file='analysis/xt.m.Rda')
load(file='analysis/xg.Rda')
```
Figure 2 compares the significant transcripts using effect and volcano plots for the first two \(\gamma\) parameters. We can see that the `r length(nsig)` transcripts are excluded by even the smallest setting of \(\gamma=0.1\). All of these have very low difference between and include the majority of the transcripts with very low dispersion.

```{r echo=FALSE, include=TRUE, out.width="0.8\\linewidth",  fig.align="center", fig.cap=c('Effect and volcano plots showing the significant transcripts for the first two \\(\\gamma\\) values of 1e-3 and 0.1. Transcripts that are significantly different with a q-value <= 0.5 with both \\(\\gamma\\) values are in red, those significant with \\(\\gamma = 0.1 \\) are in orange. Those that are not significant are in dark grey.')}


par(mfrow=c(1,2))
aldex.plot(x.sens[[1]])
points(x.sens[[1]]$diff.win[nsig], x.sens[[1]]$diff.btw[nsig], pch=19, cex=0.5, col='orange')

aldex.plot(x.sens[[1]], type='volcano')
points(x.sens[[1]]$diff.btw[nsig], -log10(x.sens[[1]]$we.eBH[nsig]), , pch=19, cex=0.5, col='orange')
```
companion to Li et al 2022 
https://rpubs.com/LiYumei/806213
Running this vs. the test data with ALDEx2 gives ~same Wilcoxon results 
they convert to cpm, I used the actual data
adding scale removed almost all significant features but the base result is 
similar
cor is ~0.97
-->

Supplementary Figure 2 shows an  analysis of synthetic data to identify the FDR and sensitivity with a standard of truth in modeled data.  This second dataset is the BRCA1 dataset from Li et al. [@Li:2022aa] which is  a transcriptome analysis of breast cancer and matched tissue controls from The Cancer Genome Atlas collection [@CGA:2013aa] with the count tables collected from the Li et al. supplementary dataset. Transcripts that were differentially abundant at different gamma values are plotted. The first panel in the top row shows FDR at different \(\gamma\) (for ALDEx2) and fold-change cutoffs (for DESeq2) and the second panel in the top row does the same for sensitivity. It is noteworthy that in this dataset the FDR for DESeq2 is still very high and that ALDEx2 behaves similarly to the PD-1 dataset. The sensitivity for both tools is much higher and adding a small amount of scale uncertainty strongly controls the FDR while having minimal impact on sensitivity.  The panels in the  bottom row show volcano and effect plots of the outputs for ALDEx2 and recapitulate the results of Figure 3 in the text. 


```{r brca_data, echo=FALSE, include=TRUE, fig.dim=c(8,5),  fig.align="center", out.width="1\\linewidth",  fig.align="center", fig.cap=c('Adding scale reduces positive identifications  in biological replicate experiments; data from the Li et al. 2022. The sensitivity analysis in the plot on the left shows that many transcripts are excluded, even when the gamma value is very low. The colors are the same as in Supplementary Figure 1. The effect plot on the left shows that those transcripts with low difference between and low dispersion are not identified when scale is included (blue) vs. when scale is not included (blue or red). The middle plot shows that those transcripts where the difference in dispersion between the scaled and unscaled analysis is largest are most affected by adding scale. This plot and the next, color transcripts identified as signficant (FDR less than 0.05) when scale is included (blue) vs. when scale is not included (blue and red). The right-hand histogram shows that the centre of the data is not on 0 fold-change between groups, and so a full scale model would likely be useful.')}

##### BRCA data from Li 2022
# an example of a bio-replicated dataset

# brca <- read.table('data-Li2022/permuted 
# datasets_TCGA/TCGA-BRCA.normal-tumor.pair.rawCount.tsv', header=T, row.names=1, 
# sep='\t')
 
# brca.conds <- as.vector(unlist(read.table('data-Li2022/permuted 
# datasets_TCGA/TCGA-BRCA.conditions.tsv', sep='\t')))

# filter using edgeR 
# from 60483 to 21813 features
# library(edgeR)
# 
#  y <- DGEList(counts=brca, group=factor(brca.conds))
#  keep <- filterByExpr(y)
#  y <- y[keep,keep.lib.sizes=FALSE]
#  brca.data <- y$counts
#  
# s.l.o.w.
# brca.sens <- aldex.senAnalysis(x, gamma=c(0,0.1,0.2,0.3,0.5,0.7))
 
# save(brca.sens, file='data-Li2022/brca.sens.Rmd')
load(file='data-Li2022/brca.sens.Rmd')

sig.orig <- which(brca.sens$gamma_0$we.eBH < 0.05)
sig.scale <- which(brca.sens$gamma_0.5$we.eBH < 0.05)

par(mfrow=c(1,3))

# which change diff.win
plot(brca.sens$gamma_0$diff.win, (brca.sens$gamma_0.5$diff.win - brca.sens$gamma_0$diff.win), pch=19, cex=0.4, col=rgb(0,0,0,0.3),xlab="base dispersion", ylab="scale - base dispersion")

points(brca.sens$gamma_0$diff.win[sig.orig], (brca.sens$gamma_0.5$diff.win - brca.sens$gamma_0$diff.win)[sig.orig], pch=19, cex=0.4, col=rgb(1,0,0,0.7))

points(brca.sens$gamma_0$diff.win[sig.scale], (brca.sens$gamma_0.5$diff.win-brca.sens$gamma_0$diff.win)[sig.scale], pch=19, cex=0.4, col=rgb(0,0,1,1))

# which change diff.btw
plot(brca.sens$gamma_0$diff.btw, (brca.sens$gamma_0.5$diff.btw-brca.sens$gamma_0$diff.btw), pch=19, cex=0.4, col=rgb(0,0,0,0.3), xlab="base logFC", ylab="scale - base logFC")

points(brca.sens$gamma_0$diff.btw[sig.orig], (brca.sens$gamma_0.5$diff.btw-brca.sens$gamma_0$diff.btw)[sig.orig], pch=19, cex=0.4, col=rgb(1,0,0,0.7))

points(brca.sens$gamma_0$diff.btw[sig.scale], (brca.sens$gamma_0.5$diff.btw-brca.sens$gamma_0$diff.btw)[sig.scale], pch=19, cex=0.4, col=rgb(0,0,1,1))

hist(brca.sens$gamma_0$diff.btw, breaks=99, xlab="logFC", main="Histogram of LFC")
abline(v=0, col='red')
par(mfrow=c(1,1))

```


The first panel in Supplementary Figure 3 shows that  the lowest dispersion transcripts are differentially affected by adding scale uncertainty. The graph in this first panel supports the concept that there is not a defined fold-change cutoff when \(\gamma=0.5\), but rather it is a dynamic between fold-change and dispersion and is consistent with the result seen in Figure 3 in the main text in a different dataset. The second panel shows that the log-fold change is only affected minimally by adding scale. The final panel shows that there is a slight scale mis-specification, in that the median difference between groups is not centered exactly at 0, and so this dataset may benefit from a full (informed) scale model. This analysis was again conducted using the BRCA1 dataset.


```{r sensitivity.plot, echo=FALSE, include=TRUE, fig.dim=c(8,5),  fig.align="center", out.width="1\\linewidth",  fig.align="center", fig.cap=c('Adding scale uncertainty reduces positive identifications  in biological replicate experiments; data from the BRCA dataset obtained from Li et al. 2022. The sensitivity analysis  plot  shows that many transcripts are excluded, even when the gamma value is very low. This is an indication that these transcripts are extremely sensitive to normalization assumptions and are likely to be FP. The colors are the same as in Supplementary Figure 1.')}
plotGamma(brca.sens, blackWhite=T)[[2]]

```

Supplementary Figure 4 shows a sensitivity analysis of the BRCA1 dataset. In the when \(\gamma=1e^{-3}\) there are `r length(which(brca.sens$gamma_0$we.eBH < 0.05))` significant transcripts (72% of all 21813 transcripts!), which drops to `r length(which(brca.sens$gamma_0.1$we.eBH < 0.05))` when \(\gamma=0.1\) and `r length(which(brca.sens$gamma_0.2$we.eBH < 0.05))` when \(\gamma=0.2\), `r length(which(brca.sens$gamma_0.3$we.eBH < 0.05))` when \(\gamma=0.3\), `r length(which(brca.sens$gamma_0.5$we.eBH < 0.05))` when \(\gamma=0.5\) and finally `r length(which(brca.sens$gamma_0.7$we.eBH < 0.05))` when \(\gamma=0.7.\). Here there is a sharp drop in significant transcripts betwee \(\gamma=0.1, 0.3\) and then relative stability between 0.3 and 0.5, suggesting that these latter \(\gamma\) values are likely to be appropriate. Note that in biologically-replicated data adding a small amount of scale uncertainty has less of an effect at small uncertainty values than in technically replicated data.
\newpage

<!--
Supplementary Figure 3 shows how dispersion behaves in linear and log space. The variance or dispersion always increases with increasing raw (or normalized) read count but decreases when measured on the log-ratio transformed data [@fernandes:2013;@Love:2014aa], reaching a minimum at some mid-point of the distribution. This makes the counter-intuitive suggestion that genes with moderate expression have more predictable expression than genes with very high expression such as housekeeping genes. This is at odds with the known biology of cells where single cell counting of housekeeping transcripts shows that they are both highly expressed and have little intrinsic variation [@Taniguchi:2010aa]. Furthermore, the dispersion is exceeding small being, for many transcripts, almost negligible. To show this point more clearly, the majority of the transcripts in the lowest decile of dispersion indicated below the dashed grey line are statistically significantly different (75\% with DESeq2, 69\% with ALDEx2), suggesting that low dispersion estimates lead to many false positives.  Indeed, benchmarking and comparison studies  repeatedly show that the choice of normalization plays a role in which results are returned as significant [@maza2013;@Dillies:2013;@Weiss:2017aa]. From the perspective of Nixon et al. [-@nixon2023scale], it is reasonable to conclude that some of these results may be due to the choice of normalization.
-->

```{r disp, eval=T, echo=F, warning=F, message=F,comment=F, fig.cap="Adding scale uncertainty changes the dispersion distribution. Panel A shows a plot of the expected value for relative abundance vs the expected value for the pooled dispersion as output by \\texttt{aldex.effect}. The dashed  horizontal lines show the median value for the features with a rAbundance between -0.5 and 0.5, and the light colored lines are lowess lines of fit through the center of mass of the data. Panel B plots the dispersion difference between  \\(\\gamma = 1\\) and \\(\\gamma = 0\\); note the non-linear relationship that highlights  the rotation that is evident in Panel A. The colored lines indicate the lowess line of fit through the centre of mass of the plot for the various  populations of points. The thick grey line is the total population and shows the difference \\(\\Delta\\), the red line is the population of significant transcripts (\\*) with \\(\\gamma = 0\\), the dashed orange line is the population of significant transcripts with a difference threshold (T) of about \\(\\pm 2^{1.4}\\)-fold change, the dashed blue line is the population of significant transcripts with \\(\\gamma = 0.5\\), and the dashed cyan line is the significant population with \\(\\gamma = 1\\). \\(\\Delta\\): Difference, *: significant, T: thresholded. "}

load(file="analysis/yst.all.Rda") 
load(file="analysis/yst.s.all.Rda") 
load(file="analysis/yst.1.all.Rda") # DESeq2 
load('analysis/res.Rda')

# with gamma = 0.5 
load('analysis/x.s.all.Rda') 
sig.des <- which(res@listData$padj < 0.05) 
sig.ald <- which(yst.all$we.eBH < 0.05) 
sig.s.ald <- which(yst.s.all$we.eBH < 0.05)

sig.all <- yst.all$we.eBH < 0.05 
sig.t <- yst.all$we.eBH < 0.05 & abs(yst.all$diff.btw) > 1.4 
sig.s <- yst.s.all$we.eBH < 0.05 
sig.1 <- yst.1.all$we.eBH < 0.05 

cuts <- yst.all$rab.all > -0.5 & yst.all$rab.all < 5 

par(mfrow=c(1,2)) 
mn.mid <- median(yst.all$diff.win[cuts]) 
mn.mid.s <- median(yst.s.all$diff.win[cuts]) 
mn.mid.1 <- median(yst.1.all$diff.win[cuts])

no.scale <- data.frame(yst.all$rab.all, yst.all$diff.win) 
half.scale <- data.frame(yst.s.all$rab.all, yst.s.all$diff.win) 
full.scale <- data.frame(yst.1.all$rab.all, yst.1.all$diff.win)

# we change dispersion a lot for the bulk of the features 
plot(yst.all$rab.all, yst.all$diff.win, pch=19, cex=0.5, col=rgb(0,0,0,0.3), xlab='rAbundance', ylab='Dispersion') 
lines(lowess(no.scale, f=0.1), col='grey', lwd=3) 
points(yst.s.all$rab.all, yst.s.all$diff.win, pch=19, cex=0.5, col=rgb(1,0,0,0.3)) 
lines(lowess(half.scale, f=0.1), col='pink', lwd=3) 
points(yst.1.all$rab.all, yst.1.all$diff.win, pch=19, cex=0.5, col=rgb(0,0,1,0.3)) 
lines(lowess(full.scale, f=0.1), col='royalblue1', lwd=3) 
abline(h=mn.mid, lty=2) 
abline(h=mn.mid.s, lty=2, col='red') 
abline(h=mn.mid.1, lty=2, col='blue') 
title(main='A', line=-1.2, adj=0.9)

g0 <- expression(paste(gamma, " = 0")) 
g5 <- expression(paste(gamma, " = 0.5")) 
g1 <- expression(paste(gamma, " = 1")) 
legend(1,4, legend=c(g0, g5, g1), pch=19, col=c("black", "red", "blue"))

# plot(yst.all$diff.win, yst.1.all$diff.win, pch=19, cex=0.5, 
# col=rgb(0,0,1,0.3), ylim=c(0,4.5), xlab='g=0 Dispersion', ylab='g=0.5 | g=1 
# Dispersion') points(yst.all$diff.win, yst.s.all$diff.win, pch=19, cex=0.5, 
# col=rgb(1,0,0,0.3)) abline(0,1, lty=2) title(main='B', line=-1.2, adj=0.5) 
# legend(0,4, legend=c( "0.5", "1"), pch=19, col=c("red", "blue"))

diff.df <- data.frame(yst.all$rab.all, yst.1.all$diff.win- yst.all$diff.win) 
sig.df <- data.frame(yst.all$rab.all[sig.all], yst.1.all$diff.win[sig.all]- yst.all$diff.win[sig.all]) 
sig.t.df <- data.frame(yst.all$rab.al[sig.t], yst.1.all$diff.win[sig.t] - yst.all$diff.win[sig.t]) 
sig.s.df <- data.frame(yst.all$rab.all[sig.s], yst.1.all$diff.win[sig.s] - yst.all$diff.win[sig.s]) 
sig.1.df <- data.frame(yst.all$rab.all[sig.1], yst.1.all$diff.win[sig.1]- yst.all$diff.win[sig.1])

ylab1 <- expression(paste("Dispersion: ", gamma, "1 - ", gamma, "0"))

plot(yst.all$rab.all, yst.1.all$diff.win- yst.all$diff.win, xlab='rAbundance', ylab=ylab1, pch=19, cex=0.5, xlim=c(-11,10), ylim=c(.1,1.5), col=rgb(0,0,0,0.8)) 
title(main='B', line=-1.2, adj=0.9) 
lines(lowess(diff.df, f=0.1), col='grey70', lwd=6) 
lines(lowess(sig.df, f=0.1), col='red', lwd=3) 
lines(lowess(sig.t.df, f=0.5), col='orange', lwd=3, lty=2) 
lines(lowess(sig.s.df, f=0.5), col='royalblue1', lwd=3, lty=2) 
lines(lowess(sig.1.df, f=0.7), col='cyan', lwd=3, lty=2)

d0 <- expression(paste(Delta)) 
dstar <- expression(paste(Delta,"*")) 
dstarT <- expression(paste(Delta,"*T")) 
dg5 <- expression(paste(Delta,"*", gamma, "0.5")) 
dg1 <- expression(paste(Delta,"*", gamma, "1"))

legend(-12,1.6, legend=c( d0, dstar, dstarT,dg5, dg1), pch="-", col=c("grey", "red", "orange", "royalblue1", "cyan"))

```


The effect on dispersion with increasing  amounts of scale uncertainty in the technical replication dataset (yeast) are shown in Supplementary Figure 5A, where we can see that the  dispersion increases as uncertainty is added. Note that the dispersion in the unscaled analysis in Figure 5A reaches a minimum near the mid-point of the distribution, and also does so when the analysis is conducted with DESeq2 (Supplementary Figure 5). This shows more clearly that dispersion of many transcripts is  almost negligible in the absence of scale uncertainty. This plot makes the counter-intuitive suggestion that the variance in expression of the majority of genes with moderate expression is more predictable  than highly-expressed genes or of housekeeping genes [@Rocha:2020aa].  This is at odds with the known biology of cells where single cell counting of highly-expressed transcripts shows that they have little intrinsic variation [@Taniguchi:2010aa;@Scott:2010].

Adding scale uncertainty by setting \(\gamma=0.5\) , or \(\gamma = 1.0\), increases the minimum dispersion as shown in Figure 5A by the red and blue data points, and by the colored lines of fit through the centre of mass of the data. Less obvious is that the additional dispersion is  not applied equally to all points. Figure 5B shows a plot of the difference between the \(\gamma= 0\) and \(\gamma= 1\) data  and here we can see that  scale uncertainty is preferentially increasing the dispersion of the mid-expressed transcripts that formerly had negligible dispersion; examine the grey line of best fit (overlaid by the red line) for the trend. Panel B also shows the trend of the expression-dispersion relationship for transcripts that are classed as statistically significant. The red line shows the trendline with no added scale uncertainty, and this trendline exactly overlays with the grey trendline for the bulk of transcripts. The orange trendline indicates those transcripts that are both statistically significant and that have a thresholded expression level of \( \pm 1.4\), and the dark blue and cyan lines show the statistically significant trendline for \(\gamma=0.5\) , or \(1.0\).

```{r dispersion, echo=F, warning=F, message=F,comment=F, result=F, cache=T,fig.dim=c(7,3), fig.cap="Plot of abundance v dispersion for the yeast  transcriptome dataset as counts,	 as logarithms of counts, and as CLR values. Panel A shows that the data are over-dispersed relative to a Poisson distribution which is represented by the dashed line when plotted on a log-log scale. Panel B shows that the relationship between the mean and the dispersion calculated in DESeq2, here the standard error (SE) of the mean, is very different when the data are log-transformed first. Panel C shows the equivalent values calculated by ALDEx2	 in which the expected CLR value for each transcipt are plotted vs. the expected dispersion. Panel D shows the output for ALDEx2 with \\( \\gamma=0.5 \\). The red line in each panel shows the LOESS line of fit to the mid-point of the distributions. In panels B and C the amount of dispersion reaches a minimum at moderate	 values.  The dashed orange line in panel A is the line of equivalence, and in panel B and C is the minimum y value. The values below the dashed grey line in panels B and C represent those below the first decile of dispersion." }


url <- "https://raw.githubusercontent.com/ggloor/datasets/main/transcriptome.tsv"
yst <- read.table(url, header=T, row.names=1)
# remove the one gene with 0 reads

yst <- yst[rownames(yst) != "YOR072W-B",]

# Gierlinski:2015aa
yst[,c('SNF2.6', 'SNF2.13','SNF2.25','SNF2.35')] <- NULL 
yst[,c('WT.21','WT.22','WT.25','WT.28','WT.34','WT.36')] <- NULL  

conds <- c(rep('S', 44), rep('W', 42))
coldata <- data.frame(conds)

# DESeq2
load('analysis/res.Rda')

#ALDEx2
load('analysis/x.all.Rda')
load('analysis/x.s.all.Rda')

# with gamma = 1
load('analysis/x.s.all.Rda')
sig.des <- which(res@listData$padj < 0.05)
sig.ald <- which(x.all$we.eBH < 0.05)
sig.s.ald <- which(x.s.all$we.eBH < 0.05)

# get and plot mean/var or mean/SE for counts
# and log counts
yst.mn <- apply(yst, 1, mean)
yst.mn.log <- apply(yst, 1, function(x) mean(log2(x)))
yst.mn.log[is.infinite(yst.mn.log)] <- 0

yst.v <- apply(yst, 1, var)

par(mfrow=c(1,4))
plot(log2(yst.mn), log2(yst.v), pch=19, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='Mean count', ylab='Variance')
#points(log2(yst.mn[sig.des]), log2(yst.v[sig.des]), pch=19, cex=0.5, col='orange')
def <- data.frame(log2(yst.mn),log2(yst.v))
lines(lowess(def, f=0.1), col=2, lwd=2)
title('A', adj=0, line= 0.8)
abline(0,1, col=rgb(1,0.6,0,0.5), lwd=2, lty=2)

l.df <- data.frame(yst.mn.log, res@listData$lfcSE)
plot(l.df, pch=19, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='Mean log2(count)', ylab='SE log2(count)',ylim=c(0,0.4))
lines(lowess(l.df, f=0.25), col=2, lwd=2)
abline(h=min(lowess(l.df)$y), col=rgb(1,0.6,0,0.5), lwd=2, lty=2)
 abline(h=0.0275, col='grey', lty=2)
title('B', adj=0, line= 0.8)

l.df <- data.frame(x.all$rab.all, x.all$diff.win)
plot(x.all$rab.all, x.all$diff.win, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='E(clr)', ylab='E(disp)')
#points(x.all$rab.all[sig.ald], x.all$diff.win[sig.ald], col='orange', cex=0.4)
lines(lowess(l.df, f=.1), col=2, lwd=2)
abline(h=min(lowess(l.df)$y), col=rgb(1,0.6,0,0.5), lwd=2, lty=2)
 abline(h=0.208, col='grey', lty=2)
title('C', adj=0, line= 0.8)

l.s.df <- data.frame(x.s.all$rab.all, x.s.all$diff.win)
plot(x.s.all$rab.all, x.s.all$diff.win, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='E(clr)', ylab='E(disp)', ylim=c(0,4.5))
#points(x.s.all$rab.all[sig.ald], x.s.all$diff.win[sig.ald], col='orange', cex=0.4)
lines(lowess(l.s.df, f=.1), col=2, lwd=2)
abline(h=min(lowess(l.s.df)$y), col=rgb(1,0.6,0,0.5), lwd=2, lty=2)
 abline(h=0.208, col='grey', lty=2)
title('D', adj=0, line= 0.8)
```

\newpage

Supplementary Figure 6 shows that the informed models with 5% or 14% difference in location between groups and \(\gamma=0.5\) provide nearly the same output for q-values, effect sizes and difference between groups (black). The line of identity is given as the grey diagonal. However, using the default CLR values to specify location are very different. In the q-value and effect plots, there is multiple populations of points that indicate the Type 1 and 2 errors that occur when the location is not specified properly. In the difference between groups plot, we see only a shift in the location that corresponds to moving the mass of the data points down by about 2.2 units (See Figure 3 in the main text)

```{r echo=FALSE, include=TRUE, fig.dim=c(7,3),  fig.align="center", fig.cap=c("Plots showing the similarity of outputs with different scale parameters. The black points show that using either a an informed model with 5% or 14% difference in location has a minimal effect on either the q-values, the effect size or difference between groups. In red, the same values  are plotted with the default model that uses a naive estimate of the location derived from the CLR.")}

par(mfrow=c(1,3))
plot(xt.m.all$we.eBH,xt.lv.all$we.eBH, pch=19, cex=0.5, col=rgb(0,0,0,0.3),
  main="q-value", log='xy', ylab="5% diff", xlab="14% diff")
points(xt.m.all$we.eBH, xg.all$we.eBH, pch=19, cex=0.5, col=rgb(1,0,0,0.3))
abline(0,1, lty=2, lwd=2, col='grey')
plot(xt.m.all$effect,xt.lv.all$effect, pch=19, cex=0.5, col=rgb(0,0,0,0.3),
  main="effect", ylab="5% diff", xlab="14% diff")
points(xt.m.all$effect, xg.all$effect, pch=19, cex=0.5, col=rgb(1,0,0,0.3))
abline(0,1, lty=2, lwd=2, col='grey')

plot(xt.m.all$diff.btw,xt.lv.all$diff.btw, pch=19, cex=0.5, col=rgb(0,0,0,0.3),
  main="diff.btw", ylab="5% diff", xlab="14% diff")
points(xt.m.all$diff.btw, xg.all$diff.btw, pch=19, cex=0.5, col=rgb(1,0,0,0.3))

abline(0,1, lty=2, lwd=2, col='grey')


```
\newpage

# References
