---
title: "Supplement: Explicit Scale Simulation for analysis of RNA-sequencing with ALDEx2
"
shorttitle: "Scale ALDEx2 Supp"
citeproc: TRUE
cite-method: citeproc
bibliography: /Users/ggloor/Library/texmf/bibtex/bib/bibdesk_refs.bib
csl: /Users/ggloor/Documents/0_git/csl_styles/nucleic-acids-research.csl
pdf-engine: latex
output:
  pdf_document
---

```{r echo=F, warning=F, message=F}
library(ALDEx2)
data(selex)
```
```{r echo=FALSE, include=TRUE, out.width="0.8\\linewidth",  fig.align="center", fig.cap=c("The aldex.senAnalysis() function was used to generate a dataset with \\(\\gamma\\) values of 1e-3, 0.1, 0.2, 0.3, 0.4,0.5, 0.7 and 1. The plotGamma() function was used to plot the result. Transcripts that are statistically significant are shown in black, and if not significant are in grey. The \\(\\gamma\\) values, standardized effect size and number of significant transcripts at each value are given on the axes.  ")}
knitr::include_graphics("analysis/scaleSim-yeast.pdf")
## plotGamma needs to be cleaned up for production
## so this figure was done by hand with the example
## snippet below - ugly sorry
#plotGamma(x.sens, blackWhite=T)
#text(6.2,345, label=length(which(x.sens[[8]]$we.eBH < 0.05)))
#text(-2,345, label=length(which(x.sens[[7]]$we.eBH < 0.05)))
#text(-6.9,345, label=length(which(x.sens[[6]]$we.eBH < 0.05)))
#text(-9.6,345, label=length(which(x.sens[[5]]$we.eBH < 0.05)))
#text(-12.2,345, label=length(which(x.sens[[4]]$we.eBH < 0.05)))
#text(-14.8,345, label=length(which(x.sens[[3]]$we.eBH < 0.05)))
#text(-17.4,345, label=length(which(x.sens[[2]]$we.eBH < 0.05)))
#text(-20,345, label=length(which(x.sens[[1]]$we.eBH < 0.05)))
#text(-6.9,355, label="N significant")
```

 One root cause of the large number of significant parts is the very low dispersion of transcripts. Figure 1 shows a graphical output from the texttt{aldex.scaleSim()} function with the yeast transcriptome dataset described in the text [@Gierlinski:2015aa;@Schurch:2016aa]. This allows us to examine which transcripts are sensitive to even minimal amounts of scale uncertainty using \(\gamma = (1e-3, 0.1, 0.2, 0.3, 0.4,0.5, 0.7, 1) \). Here it is obvious that even a negligible amount of scale uncertainty removes over half of the transcripts that were formerly significantly different, and all of these had very small effect sizes. 

We recommend a minimum scale uncertainty of 0.5, and suggest that the texttt{aldex.senAnalysis()} function be run on all analyses. The individual \texttt{aldex()} outputs can be accessed as sequential entries in the list output, or the analysis as a whole can be plotted with the \texttt{plotGamma} function.

```{r, echo=F, warning=F, message=F,comment=F, result=F} 
library(DESeq2)
library(ALDEx2)
#ALDEx2
load(file="analysis/yst.all.Rda")
load(file="analysis/yst.s.all.Rda")
load(file="analysis/yst.1.all.Rda")
# with gamma = 0.5
load('analysis/x.s.all.Rda')
load('analysis/yst.sens.Rda')
nsig <- x.sens[[2]]$we.eBH > 0.05 & x.sens[[1]]$we.eBH < 0.05

# data to compare different scale means
load(file='analysis/xt.lv.Rda')
load(file='analysis/xt.m.Rda')
load(file='analysis/xg.Rda')
```

Figure 2 compares the significant transcripts using effect and volcano plots for the first two \(\gamma\) parameters. We can see that the `r length(nsig)` transcripts are excluded by even the smallest setting of \(\gamma=0.1\). All of these have very low difference between and include the majority of the transcripts with very low dispersion.




```{r echo=FALSE, include=TRUE, out.width="0.8\\linewidth",  fig.align="center", fig.cap=c('Effect and volcano plots showing the significant transcripts for the first two \\(\\gamma\\) values of 1e-3 and 0.1. Transcripts that are significantly different with a q-value <= 0.5 with both \\(\\gamma\\) values are in red, those significant with \\(\\gamma = 0.1 \\) are in orange. Those that are not significant are in dark grey.')}


par(mfrow=c(1,2))
aldex.plot(x.sens[[1]])
points(x.sens[[1]]$diff.win[nsig], x.sens[[1]]$diff.btw[nsig], pch=19, cex=0.5, col='orange')

aldex.plot(x.sens[[1]], type='volcano')
points(x.sens[[1]]$diff.btw[nsig], -log10(x.sens[[1]]$we.eBH[nsig]), , pch=19, cex=0.5, col='orange')
```
\newpage

Figure 3 shows how dispersion behaves in linear and log space. The variance or dispersion always increases with increasing raw (or normalized) read count but decreases when measured on the log-ratio transformed data [@fernandes:2013;@Love:2014aa], reaching a minimum at some mid-point of the distribution. This makes the counter-intuitive suggestion that genes with moderate expression have more predictable expression than genes with very high expression such as housekeeping genes. This is at odds with the known biology of cells where single cell counting of housekeeping transcripts shows that they are both highly expressed and have little intrinsic variation [@Taniguchi:2010aa]. Furthermore, the dispersion is exceeding small being, for many transcripts, almost negligible. To show this point more clearly, the majority of the transcripts in the lowest decile of dispersion indicated below the dashed grey line are statistically significantly different (75\% with DESeq2, 69\% with ALDEx2), suggesting that low dispersion estimates lead to many false positives.  Indeed, benchmarking and comparison studies  repeatedly show that the choice of normalization plays a role in which results are returned as significant [@maza2013;@Dillies:2013;@Weiss:2017aa]. From the perspective of Nixon et al. [-@nixon2023scale], it is reasonable to conclude that some of these results may be due to the choice of normalization.


```{r dispersion, echo=F, warning=F, message=F,comment=F, result=F, cache=T,fig.dim=c(7,3), fig.cap="Plot of abundance v dispersion for the yeast  transcriptome dataset as counts,	 as logarithms of counts, and as CLR values. Panel A shows that the data are over-dispersed relative to a Poisson distribution which is represented by the dashed line when plotted on a log-log scale. Panel B shows that the relationship between the mean and the dispersion calculated in DESeq2, here the standard error (SE) of the mean, is very different when the data are log-transformed first. Panel C shows the equivalent values calculated by ALDEx2	 in which the expected CLR value for each transcipt are plotted vs. the expected dispersion. Panel D shows the output for ALDEx2 with \\( \\gamma=0.5 \\). The red line in each panel shows the LOESS line of fit to the mid-point of the distributions. In panels B and C the amount of dispersion reaches a minimum at moderate	 values.  The dashed orange line in panel A is the line of equivalence, and in panel B and C is the minimum y value. The values below the dashed grey line in panels B and C represent those below the first decile of dispersion." }


url <- "https://raw.githubusercontent.com/ggloor/datasets/main/transcriptome.tsv"
yst <- read.table(url, header=T, row.names=1)
# remove the one gene with 0 reads

yst <- yst[rownames(yst) != "YOR072W-B",]

# Gierlinski:2015aa
yst[,c('SNF2.6', 'SNF2.13','SNF2.25','SNF2.35')] <- NULL 
yst[,c('WT.21','WT.22','WT.25','WT.28','WT.34','WT.36')] <- NULL  

conds <- c(rep('S', 44), rep('W', 42))
coldata <- data.frame(conds)

# DESeq2
load('analysis/res.Rda')

#ALDEx2
load('analysis/x.all.Rda')
load('analysis/x.s.all.Rda')

# with gamma = 1
load('analysis/x.s.all.Rda')
sig.des <- which(res@listData$padj < 0.05)
sig.ald <- which(x.all$we.eBH < 0.05)
sig.s.ald <- which(x.s.all$we.eBH < 0.05)

# get and plot mean/var or mean/SE for counts
# and log counts
yst.mn <- apply(yst, 1, mean)
yst.mn.log <- apply(yst, 1, function(x) mean(log2(x)))
yst.mn.log[is.infinite(yst.mn.log)] <- 0

yst.v <- apply(yst, 1, var)

par(mfrow=c(1,4))
plot(log2(yst.mn), log2(yst.v), pch=19, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='Mean count', ylab='Variance')
#points(log2(yst.mn[sig.des]), log2(yst.v[sig.des]), pch=19, cex=0.5, col='orange')
def <- data.frame(log2(yst.mn),log2(yst.v))
lines(lowess(def, f=0.1), col=2, lwd=2)
title('A', adj=0, line= 0.8)
abline(0,1, col=rgb(1,0.6,0,0.5), lwd=2, lty=2)

l.df <- data.frame(yst.mn.log, res@listData$lfcSE)
plot(l.df, pch=19, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='Mean log2(count)', ylab='SE log2(count)',ylim=c(0,0.4))
lines(lowess(l.df, f=0.25), col=2, lwd=2)
abline(h=min(lowess(l.df)$y), col=rgb(1,0.6,0,0.5), lwd=2, lty=2)
 abline(h=0.0275, col='grey', lty=2)
title('B', adj=0, line= 0.8)

l.df <- data.frame(x.all$rab.all, x.all$diff.win)
plot(x.all$rab.all, x.all$diff.win, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='E(clr)', ylab='E(disp)')
#points(x.all$rab.all[sig.ald], x.all$diff.win[sig.ald], col='orange', cex=0.4)
lines(lowess(l.df, f=.1), col=2, lwd=2)
abline(h=min(lowess(l.df)$y), col=rgb(1,0.6,0,0.5), lwd=2, lty=2)
 abline(h=0.208, col='grey', lty=2)
title('C', adj=0, line= 0.8)

l.s.df <- data.frame(x.s.all$rab.all, x.s.all$diff.win)
plot(x.s.all$rab.all, x.s.all$diff.win, cex=0.5, col=rgb(0,0,0,0.2),
  xlab='E(clr)', ylab='E(disp)', ylim=c(0,4.5))
#points(x.s.all$rab.all[sig.ald], x.s.all$diff.win[sig.ald], col='orange', cex=0.4)
lines(lowess(l.s.df, f=.1), col=2, lwd=2)
abline(h=min(lowess(l.s.df)$y), col=rgb(1,0.6,0,0.5), lwd=2, lty=2)
 abline(h=0.208, col='grey', lty=2)
title('D', adj=0, line= 0.8)
```

\newpage

Figure 4 shows that the informed models with 5% or 14% difference in location between groups and \(\gamma=0.5\) provide nearly the same output for q-values, effect sizes and difference between groups (black). The line of identity is given as the grey diagonal. However, using the default CLR values to specify location are very different. In the q-value and effect plots, there is multiple populations of points that indicate the Type 1 and 2 errors that occur when the location is not specified properly. In the difference between groups plot, we see only a shift in the location that corresponds to moving the mass of the data points down by about 2.2 units (See Figure 3 in the main text)

```{r echo=FALSE, include=TRUE, fig.dim=c(7,3),  fig.align="center", fig.cap=c("Plots showing the similarity of outputs with different scale parameters. The black points show that using either a an informed model with 5% or 14% difference in location has a minimal effect on either the q-values, the effect size or difference between groups. In red, the same values  are plotted with the default model that uses a naive estimate of the location derived from the CLR.")}

par(mfrow=c(1,3))
plot(xt.m.all$we.eBH,xt.lv.all$we.eBH, pch=19, cex=0.5, col=rgb(0,0,0,0.3),
  main="q-value", log='xy', ylab="5% diff", xlab="14% diff")
points(xt.m.all$we.eBH, xg.all$we.eBH, pch=19, cex=0.5, col=rgb(1,0,0,0.3))
abline(0,1, lty=2, lwd=2, col='grey')
plot(xt.m.all$effect,xt.lv.all$effect, pch=19, cex=0.5, col=rgb(0,0,0,0.3),
  main="effect", ylab="5% diff", xlab="14% diff")
points(xt.m.all$effect, xg.all$effect, pch=19, cex=0.5, col=rgb(1,0,0,0.3))
abline(0,1, lty=2, lwd=2, col='grey')

plot(xt.m.all$diff.btw,xt.lv.all$diff.btw, pch=19, cex=0.5, col=rgb(0,0,0,0.3),
  main="diff.btw", ylab="5% diff", xlab="14% diff")
points(xt.m.all$diff.btw, xg.all$diff.btw, pch=19, cex=0.5, col=rgb(1,0,0,0.3))

abline(0,1, lty=2, lwd=2, col='grey')


```
\newpage

# References
